examples/output_voltage_measurement/measurement.py

```diff
@@ -176,25 +176,14 @@ def _get_session_info_for_pin(session_info, pin_name):
     return session_info[session_index]
 
 
-def _add_cancel_callback(session, measurement_service, pending_cancellation):
-    def cancel_callback():
-        session_to_abort = session
-        if session_to_abort is not None:
-            nonlocal pending_cancellation
-            pending_cancellation = True
-            session_to_abort.abort()
-
-    measurement_service.context.add_cancel_callback(cancel_callback)
-
-
-def _wait_for_source_complete_event(measurement_service, channels, pending_cancellation):
+def _wait_for_source_complete_event(measurement_service, channels, cancellation_event):
     deadline = time.time() + measurement_service.context.time_remaining
     while True:
         if time.time() > deadline:
             measurement_service.context.abort(
                 grpc.StatusCode.DEADLINE_EXCEEDED, "deadline exceeded"
             )
-        if pending_cancellation:
+        if cancellation_event.wait(0.1):
```

Comment:
The purpose of this function is to wait for the source complete event. Ideally, we would call `channels.wait_for_event()` once with a long timeout, but there is no way to cancel this long wait, so we break it up into shorter 100 ms waits and check for cancellation after each short wait.

Adding a second wait to the loop hurts performance because it will keep waiting for up to 100 ms after the source complete event has occurred.

`channels.wait_for_event()` performs a synchronous wait and doesn't support multiplexed I/O like `select()` or `WaitForMultipleObjects()`, so the best solution is to call `channels._wait_for_event()` with a 100 ms timeout and use the cancellation event as an atomic flag.

```suggestion
        if cancellation_event.is_set():
```

Reply:
Thanks for the suggestion. Updated to invoke `is_set()`.